src/adagrams.js

```diff
@@ -1,15 +1,128 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const SCORE_DICT = {
+  1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+  2: ["D", "G"],
+  3: ["B", "C", "M", "P"],
+  4: ["F", "H", "V", "W", "Y"],
+  5: ["K"],
+  8: ["J", "X"],
+  10: ["Q", "Z"],
+};
+const choose_index = (choices) => {
+  let index = Math.floor(Math.random() * choices.length);
+  return choices[index];
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let random_letter = [];
+  let is_valid = true;
+  let count_obj = {};
+  while (random_letter.length < 10) {
+    ///might not work because the function was written for an array, not an object
+    let letter = choose_index(Object.keys(LETTER_POOL));
```

Comment:
Actually, upon reflecting more, I think that this might have a subtle bug where the odds of drawing an "E" isn't actually higher than the odds of drawing a "J" or an "X".

(You won't draw too many but the distribution will be subtly off.)